### PR TITLE
[Merged by Bors] - Modify wait time in waitForFirstATX

### LINKED ISSUE #3698: A node joining late might not be able to publish ATX
## Description
Sometimes a newly added go-spacemesh node, which enters the network after 2nd epoch (when other nodes have already created ATXes) is unable to create its own ATX. 

The `waitForFirstAtx()` no longer works properly because the time a new ATX is published shifted in https://github.com/spacemeshos/go-spacemesh/pull/3603. Node now waits till _poet proof deadline_ to collect proofs from many PoET services. The moment `waitForFirstAtx()` finishes the ATX is not yet published by other nodes.

In effect, the node builds its own NIPost challenge based on a stale Positioning ATX, which fails (because the publish epoch of nipost challenge is the current epoch - the poet registration window has already passed an epoch ago).

## Steps to reproduce
Run the `TestAddNodes` system test.

## Actual Behavior
Sometimes test fails

## Expected Behavior
The test should always pass

## Environment
CI, systests

## Additional Resources 
- [Grafana query from a failed run](https://ci-grafana.spacemesh.io/explore?orgId=1&left=%7B%22datasource%22:%22Loki%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22test-tvex%5C%22,%20pod%3D%5C%22smesher-27-0%5C%22%7D%20%7C~%20%60converting%20genesis%20time%20to%20local%20time%7Cchallenge%20submitted%20to%20poet%20proving%20service%7Cfailed%20to%20build%20NIPost%7Cstored%20poet%20proof%7CSelected%20the%20best%20PoET%20proof%7Cbuilding%20NIPost%7Cnew%20atx%20challenge%20is%20ready%7Cready%20to%20build%20first%20atx%7CDiscarding%20challenge%7CSetting%20up%20poet%20retry%20timer%7CUnknown%20error%7Cusing%20existing%20atx%20challenge%7Cbuilding%20new%20atx%20challenge%7Ccalculated%20poet%20proof%20deadline%20is%20too%20soon%60%22,%22editorMode%22:%22builder%22,%22queryType%22:%22range%22%7D%5D,%22range%22:%7B%22from%22:%221666848708002%22,%22to%22:%221666852308002%22%7D%7D)
- [Explore-logs-2022-10-31 16_05_41.txt](https://github.com/spacemeshos/go-spacemesh/files/9902353/Explore-logs-2022-10-31.16_05_41.txt)
- [debugging conversation on Slack](https://spacemesh.slack.com/archives/C89QJJY3G/p1666855504630539)

### PR DESCRIPTION
**Pull request by poszu**
## Motivation
Closes #3698 - see it for details.

### Poet proof deadline adjusted
Previously the deadline to wait for poet proofs was set as late as possible. As a result, the nodes created their ATXes very late (close to the start of the next poet round). It caused problems for the nodes that are new to the network.

This PR shift the deadline to the earliest possible moment which is "poet round end" + grace period.

## Time to wait for the first ATX extended
Nodes that just joined the network need a valid, "fresh", Positioning ATX targeting the current epoch. At the start, they wait a specific time to obtain it. This PR extends this wait time to wait for Poet round END + 2 * grace period + "average post duration" (hardcoded).

:bulb: I'm not quite sure about the hardcoded average post duration. It's meant to accommodate the time that other nodes need to spend generating a PoST proof to build their ATXes. I'm not sure what value it should have nor if it is really needed. 2*Grace period should be enough anyway.

## Test Plan
systests (especially `TestAddNodes`)

## DevOps Notes
- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
Bors try

File: activation/activation.go [hunk 1]
```diff
@@ -348,28 +341,40 @@ func (b *Builder) waitForFirstATX(ctx context.Context) bool {
...
+	//
+	//                 Grace ──────┐
+	//                 Period      │  ┌─ATX arrives
+	//                     │  PoST │  │   ┌ wait deadline
+	//                     │ ┌───► │  │   │   ┌ Next round start
+	//   ┌────────────────┬┴─►   └─┴─►│   │   ▼───────
+	//   │  POET ROUND N  │           │   │   │ ROUND N+1
+	// ──┴────────────┬───┴───────────▼───▼───┴───────►time
+	//     EPOCH N    │      EPOCH N+1
+	// ───────────────┴───────────────────────────────
```
Comment (dshulyak) [inline activation/activation.go:363]: what are you using to draw this stuff?

**Review by dshulyak — APPROVED** [review]

File: activation/activation.go (hunk 1 above)
Comment (poszu) [reply activation/activation.go:363]: https://asciiflow.com

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors merge

**Pull request closed without merging**